certinel.go

```diff
@@ -32,21 +33,40 @@ func New(w Watcher, errBack func(error)) *Certinel {
 
 	return &Certinel{
 		watcher: w,
+		done:    make(chan struct{}),
 		errBack: errBack,
 	}
 }
 
 // Watch setups the Certinel's channel handles and calls Watch on the
 // held Watcher instance.
 func (c *Certinel) Watch() {
+	select {
+	case <-c.done:
+		panic("attempting watch after close")
+	default:
+	}
+
 	go func() {
+		defer func() {
+			close(c.done)
```

Comment:
You could still have two calls to Watch, which would both eventually close the done channel. #7

Reply:
That is intended to be against the API.